pkg/serviceprovider/github/downloadfilecapability.go

```diff
@@ -80,3 +75,23 @@ func (f downloadFileCapability) DownloadFile(ctx context.Context, repoUrl, filep
 	}
 	return content, nil
 }
+
+func (f downloadFileCapability) parseOwnerAndRepoFromUrl(ctx context.Context, url string) (owner string, repo string, err error) {
+	urlRegexp, err := regexp.Compile(`(?Um)^` + regexp.QuoteMeta(f.githubBaseUrl) + `/(?P<owner>[^/]+)/(?P<repo>[^/]+)(/|(.git)?)$`)
```

Comment:
the reqex could be precompiled  and stored as a field of the capability instead of the base url..

Reply:
It is possible. I will define a constructor function for the capability struct.